METHOD_NAMING_PROPOSAL.md

```diff
@@ -0,0 +1,412 @@
+
+## Integration Comparison
+
+This document provides alternative naming options in comparison to the v4 integration to 
+determine the best pattern for v5. 
+
+In v4, most payment flows require one method call (ex: `tokenize`), and the browser/app switch is handled under the hood by the SDK, before returning a `PaymentMethodNonce` to the merchant's listener. Some payment methods require two method calls (ex: 3DS `performVerification` -> `continuePerformVerification`).
+
+In v5, all app/broswer switching payment methods require starting the payment flow by invoking a method on the `Client` to retrieve data to be passed to the `Launcher`. With that data, the `Launcher` presents the payment sheet, launches the browser, or launches the exteral app for some user interaction/authentication/payment authorization. The `Launcher` will be instantiated with a callback that will be invoked upon return to the merchant app from the browser/external app/payment sheet. Within the callback, another method on the `Client` will be invoked to handle data from the app/browser switch result and return a `PaymentMethodNonce`. 
+
+### Minimal Change
+
+With this implementation, the method names from v4 are maintained where possible. The comments above each method show what the method actually does, as the naming doesn't entirely align with this approach.
+
+<table>
+<tr>
+<th>v4 PayPal</th>
+<th>v5 PayPal</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to launch browser
+// Launch browser flow, parse browser result
+// Return nonce to listener
+payPalClient.tokenizePayPalAccount(activity, request)
+```
+
+</td>
+
+<td>
+
+```kotlin
+payPalLauncher = PayPalLauncher() { payPalBrowserSwitchResult ->
+  // Parse the result from the PayPal browser flow and return a nonce
+  payPalClient.onBrowserSwitchResult(payPalBrowserSwitchResult) 
+}
+
+// Retrieve data required to launch the PayPal flow in a browser
+payPalClient.tokenizePayPalAccount(context, request) { payPalResponse, error ->
+  // Launch the PayPal in a browser
+  payPalLauncher.launch(activity, payPalResponse) 
+}
+```
+
+</td>
+
+</tr>
+
+<tr>
+<th>v4 Local Payment</th>
+<th>v5 Local Payment</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to start local payment authorization
+// Or return result if no additional auth required
+localPaymentClient.startPayment(activity, request) { localPaymentResult, error ->
+  // Initiate browser flow and return nonce to listener
+  localPaymentClient.approveLocalPayment(activity, localPaymentResult)
+}
+```
+
+</td>
+
+<td>
+
+```kotlin
+localPaymentLauncher = LocalPaymentLauncher() { localPaymentBrowserSwitchResult ->
+  // Parse the result from the local payment browser flow and return a nonce
+  localPaymentClient.approveLocalPayment(localPaymentBrowserSwitchResult) 
+}
+
+// Retrieve data required to launch the local payment flow in a browser
+localPaymentClient.startPayment(context, request) { localPaymentResult, error ->
```

Comment:
this pattern is a little confusing as an Android outsider, as localPaymentClient is calling itself within the startPayment function via localPaymentLauncher. What is the localPaymentLauncher doing in this workflow that localPaymentClient can do on its own?

Reply:
Is it the activity access and management that makes the difference?

Reply:
Yes see comment above - the reason for the Launcher is to separate out the portion of the flow that needs to be aware of Android Activity (launching apps/browsers) and the portion that doesn't (API calls/tokenizing), stemming from the Android API requirement to register activity result handlers in the OnCreate method of your Activity (before being able to fetch a client token). But open to suggestions for making this pattern more clear!